WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -68,12 +68,20 @@ def index
       params[:event_ids] = params[:event_ids].values
     end
 
+    @delegates = User.delegates
+
     @past_selected = params[:state] == "past"
     @present_selected = params[:state] == "present"
     @recent_selected = params[:state] == "recent"
 
     @years = ["all years"] + Competition.where(showAtAll: true).pluck(:year).uniq.select { |y| y <= Date.today.year }.sort!.reverse!
-    @competitions = Competition.includes(:country).where(showAtAll: true).order(:year, :month, :day)
+
+    if params[:delegate].present?
+      delegate = User.find(params[:delegate])
+      @competitions = delegate.delegated_competitions.where(showAtAll: true).order(:year, :month, :day)
+    else
+      @competitions = Competition.includes(:country).where(showAtAll: true).order(:year, :month, :day)
```

Comment:
This is concerning. Why do we `includes(:country)` here, but not in the above? Won't that cause problems? We also shouldn't have to repeat the `.where(showAtAll: true).order(:year, :month, :day)` part.

I think we can DRY this up as follows:

```ruby
    if params[:delegate].present?
       delegate = User.find(params[:delegate])
       @competitions = delegate.delegated_competitions
     else
       @competitions = Competition
     end
     @competitions = @competitions.includes(:country).where(showAtAll: true).order(:year, :month, :day)

```